### PR TITLE
move IsValidCandidateName to action package

### PR DESCRIPTION
**Pull request by millken**
# Description
`action.IsValidCandidateName` as common funcs

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: action/validations.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	if len(s) == 0 || len(s) > 12 {
+		return false
+	}
+	for _, c := range s {
+		if !(('a' <= c && c <= 'z') || ('0' <= c && c <= '9')) {
+			return false
+		}
+	}
+	return true
+}
```
Comment (dustinxie) [inline action/validations.go:29]: just add this func to `candidate_register.go`?

File: action/validations_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+			input:  "",
+			output: false,
+		},
+	}
+
+	for _, tt := range tests {
+		output := IsValidCandidateName(tt.input)
+		assert.Equal(t, tt.output, output)
+	}
+}
```
Comment (dustinxie) [inline action/validations_test.go:58]: just add this to `candidate_register_test.go? `

**Review by dustinxie — APPROVED** [review]

File: action/candidateregister_test.go [hunk 3]
```diff
@@ -146,3 +147,48 @@ func TestCandidateRegisterABIEncodeAndDecode(t *testing.T) {
...
+		},
+		{
+			input:  "",
+			output: false,
+		},
+	}
+
+	for _, tt := range tests {
+		output := IsValidCandidateName(tt.input)
+		assert.Equal(t, tt.output, output)
```
Comment (dustinxie) [inline action/candidateregister_test.go:192]: use `require.Equal`, so no need to import `assert` package above

File: action/protocol/staking/validations.go [hunk 4]
```diff
@@ -17,7 +17,6 @@ import (
 // Errors
 var (
 	ErrInvalidAmount       = errors.New("invalid staking amount")
```
Comment (dustinxie) [inline action/protocol/staking/validations.go:19]: also replace this with `action.ErrInvalidAmount`

Comment (millken) [reply action/protocol/staking/validations.go]: done

**Review by huof6829 — APPROVED** [review]

**Pull request merged**